examples/next-js-netlify/Earthfile

```diff
@@ -0,0 +1,37 @@
+VERSION 0.6
+FROM --platform=linux/amd64 node:latest
+WORKDIR /app
+
+deps:
+    ARG auth_token
+    ARG site_id
+    # Set environment variables required for building and deploying to Netlify
+    ENV NETLIFY_AUTH_TOKEN=$auth_token
+    ENV NETLIFY_SITE_ID=$site_id
+    # Install the Netlify CLI
+    RUN npm install netlify-cli -g
+    # Copy files required for building and deploying to Netlify
+    COPY netlify.toml ./
+
+build:
+    FROM +deps
+    # Copy files required for building
+    COPY next-env.d.ts package.json styles.module.css tsconfig.json ./
+    # Copy directories required for building
+    COPY --dir pages ./
+    # Install required packages
+    RUN npm install netlify-cli --save-dev
+    RUN npm install @netlify/plugin-nextjs --save-dev
+    RUN npm install
```

Comment:
Similar to the above change you made - this means a change to anything in `pages` will break cache and require a full install. 

Is it possible to put all of your `npm install` commands into `deps` and then do the majority of your other commands later? 

The way I like to think about it is "put the commands with content that is most likely to change below other commands that are unlikely to change".

Reply:
This is actually the correct order of operations for a Next.js site. All of those installs have to run after files are copied. They install packages that are necessary to the site and defined in the copied package.json file.

Reply:
Oh - does that also include the pages directory? 
I would have thought it would only have been `package.json` and `package-lock.json` that are required for an install. 🤔

Reply:
Actually, you are correct. It just needs `package.json` to install dependencies. I updated the Earthfile and Readme to do all installs on the +deps target.

Reply:
Awesome 🥳